src/transformers/modeling_tf_bart.py

```diff
@@ -0,0 +1,1183 @@
+    position ids are passed to the forward function.
+    """
+
+    def __init__(self, num_embeddings: int, embedding_dim: int, padding_idx: int, offset, **kwargs):
+        # Bart is set up so that if padding_idx is specified then offset the embedding ids by 2
+        # and adjust num_embeddings appropriately. Other models dont have this hack
+        self.offset = offset
+        assert padding_idx is not None
+        num_embeddings += offset
+        super().__init__(num_embeddings, embedding_dim, **kwargs)
+
+    def call(self, input_ids: tf.Tensor, use_cache=False):
+        """Input is expected to be of size [bsz x seqlen]."""
+        bsz, seq_len = input_ids.shape[:2]
+
+        if use_cache:
+            positions = tf.fill((1, 1), seq_len - 1)
+        else:
+            # starts at 0, ends at 1-seq_len
+            positions = tf.range(0, seq_len, delta=1, dtype=tf.int32, name="range")
+        return super().call(positions + self.offset)  # super object is not callable for some reason
+
+
+@add_start_docstrings(
+    "The bare BART Model outputting raw hidden-states without any specific head on top.",
+    BART_START_DOCSTRING,
+)
+@keras_serializable
+class TFBartModel(TFPretrainedBartModel):
+    def __init__(self, config: BartConfig, *inputs, **kwargs):
+        super().__init__(config, *inputs, **kwargs)
+        self.shared = TFSharedEmbeddings(config.vocab_size, config.d_model, config.pad_token_id, name="model.shared")
+
+        with tf.compat.v1.variable_scope("model.shared") as shared_abs_scope_name:
+            pass
+
+        embed_tokens = _NoLayerEmbedTokens(self.shared, abs_scope_name=shared_abs_scope_name)
+        embed_tokens.vocab_size = self.shared.vocab_size
+        embed_tokens.hidden_size = self.shared.hidden_size
+
+        self.encoder = TFBartEncoder(config, embed_tokens, name="encoder")
+        self.decoder = TFBartDecoder(config, embed_tokens, name="decoder")
+
+    def _prepare_bart_decoder_inputs(
+        self,
+        inputs,
+        decoder_input_ids=None,
+        decoder_attn_mask=None,
+        mask_dtype=None,
+    ):
+        """Prepare masks that ignore padding tokens  decoder and a causal lm mask for the decoder if
+        none are provided. This mimics the default behavior in fairseq. To override it pass in masks.
+        """
+        pad_token_id = self.config.pad_token_id
+        if decoder_input_ids is None:
+            decoder_input_ids = self._shift_right(inputs)
+        bsz, tgt_len = decoder_input_ids.shape[:2]
+        if decoder_attn_mask is None:
+            decoder_padding_mask = make_padding_mask(decoder_input_ids, pad_token_id)
+        else:
+            decoder_padding_mask = invert_mask(tf.Tensor)
+
+        causal_lm_mask = causal_attention_mask(tgt_len, tgt_len, mask_dtype)
+        return decoder_input_ids, decoder_padding_mask, causal_lm_mask
+
+    @add_start_docstrings_to_callable(BART_INPUTS_DOCSTRING)
```

Comment:
This should also have the output docstrings part. Can be copied from the Bart modeling file.